the_pile/utils.py

```diff
@@ -1,49 +1,24 @@
 import re
 import os
 import hashlib
-from concurrent_iterator.thread import Producer
 from functools import reduce
 import operator
 import collections
-import urllib.request
 from pathlib import Path
-import gdown
 import tarfile
-import requests
 import shutil
+
+import gdown
 from tqdm import tqdm
+from best_download import download_file
 
 def touch(x):
     Path(x).touch()
 
-def download_file(url, to):
-    # modified from https://stackoverflow.com/a/37573701
-    print('Downloading {}'.format(url))
-
-    response = requests.get(url, stream=True)
-    size = int(response.headers.get('content-length', 0))
-    block_size = 1024*1024
-    pbar = tqdm(total=size, unit='iB', unit_scale=True)
-    with open(to, 'wb') as fout:
-        for data in response.iter_content(block_size):
-            pbar.update(len(data))
-            fout.write(data)
-    pbar.close()
-    assert not (size != 0 and pbar.n != size)
-
 Source = collections.namedtuple('Source', ['type', 'url'])
 
 def download(fname, checksum, sources, extract=False):
     if os.path.exists(fname + '.done'): return
-    if os.path.exists(fname):
-        try:
-            print(fname, 'already exists, verifying checksum')
-            sha256sum(fname, expected=checksum)
-            touch(fname + '.done')
-            return
-        except AssertionError:
-            print('{} exists but doesn\'t match checksum!'.format(fname))
-            rm_if_exists(fname)
```

Comment:
I think we still want to keep this; this is for if you downloaded the file already. Ignore this comment if your download code checks if existing file matches checksum already.

Reply:
Yeah, I moved this bit into gdrive only because best-download handles all the checksumming, checkpointing, checking for existing files etc.